From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 18 Jul 2026 13:51:41 +0200
Subject: arm64: dts: allwinner: sun50i-a64-pinephone: Keep Pinephone WiFi
 powered in suspend

The WiFi chip is on an always-on rail and has to keep its state across
suspend for wakeup to have anything to wake from, so tell the MMC core
not to power the slot down.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
@@ -778,6 +778,7 @@ &mmc1 {
 	non-removable;
 	post-power-on-delay-ms = <1>; /* wifi power is always on */
 	status = "okay";
+	keep-power-in-suspend;
 
 	rtl8723cs: wifi@1 {
 		reg = <1>;
-- 
Armbian

